### ISSUE TITLE
Task: Torch implementation of ledoit-wolf

### ISSUE THREAD

**Issue by g-simmons** [g-simmons/persona-research-internship#258]
## Background
*relevant information and motivation for this task*
Ledoit Wolf is a method for estimating covariance matrices

There is an implementation in sci-kit learn
https://scikit-learn.org/stable/modules/generated/sklearn.covariance.LedoitWolf.html

Ledoit Wolf is used in the calculation for LDA (Linear Discriminant Analysis). 

Most of the calculation for LDA is torch-compatible, meaning we could run it on GPU. But Ledoit-Wolf doesn't have a torch-compatible implementation, at least not in scikit-learn. 

There are existing implementations of ledoit wolf:
https://github.com/WLM1ke/LedoitWolf/blob/master/ledoit_wolf.py
https://github.com/scikit-learn/scikit-learn/blob/da08f3d99/sklearn/covariance/_shrunk_covariance.py#L467

## Task
*high-level task description*
Write a torch-compatible version of ledoit-wolf covariance estimation. Compare it to the sklearn version for speed and accuracy.

## How To
*specific guidance on implementation*

## Undecided
*implementation details that haven't been decided yet ("known-unknown" status)*

## Expected Results
*what results are expected from completing this issue*
## Dependencies
*does anything else need to happen before this issue can be completed?*
